copyspecial.py

```diff
@@ -16,29 +16,60 @@
 import argparse
 
 # This is to help coaches and graders identify student assignments
-__author__ = "???"
+__author__ = "knmarvel with madarp"
 
 
-# +++your code here+++
-# Write functions and modify main() to call them
+def get_special_paths(dir):
+    """takes a directory and returns a list of the absolute
+    paths of the special files in the given directory"""
+    spec_paths = []
+    for root, dirs, files in os.walk(dir):
+        for name in files:
+            if re.search("__(.+?)__", name):
+                if os.path.join(os.path.abspath(dir), name) not in spec_paths:
+                    spec_paths.append(os.path.join(os.path.abspath(dir), name))
+                else:
+                    return "ERROR"
+    return spec_paths
 
-def main():
-    # This snippet will help you get started with the argparse module.
+
+def copy_to(paths, dir):
+    """given a list of the paths, copy those files into the
+    given directory"""
+    for file in paths:
+        shutil.copy(file, dir)
+
+
+def zip_to(paths, dir):
+    """given a list of paths, zip those files up into the
+    given zipfile"""
+    cmd = ["zip", "-j", dir] + paths
```

Comment:
i think the assessment instructions specifically say we have to warn that we're about to execute this particular function: print("about to execute: ", cmd)